main.go

```diff
@@ -126,11 +130,17 @@ func uploadFile(api *CacophonyAPI, filename string) error {
 	}
 	defer f.Close()
 	br := bufio.NewReader(f)
-	if err := api.UploadThermalRaw(info, br); err != nil {
-		return err
+	for i := 1; i <= 3; i++ {
+		if err := api.UploadThermalRaw(info, br); err != nil {
+			log.Printf("upload failed, trying again")
```

Comment:
On the last failure before giving up "trying again" isn't true. You could have `i` count down so that it represents the remaining retries and include the remaining retry count in the log message.